### PR TITLE
Improve examples

### PR DESCRIPTION
**Pull request by mfeurer**
* demonstrate how to access the best model (#872)
* demonstrate using a custom splitter (#498)
* show how to access training accuracy (#670)

### CODE REVIEW

File: examples/40_advanced/example_resampling.py [hunk 1]
```diff
@@ -89,26 +90,39 @@
...
-    resampling_strategy_arguments={'folds': 5},
+    resampling_strategy=resampling_strategy,
+    resampling_strategy_arguments=resampling_strategy_arguments,
 )
 automl.fit(X_train, y_train, dataset_name='breast_cancer')
 
 ############################################################################
-# Get the Score of the final ensemble
-# ===================================
+# For unknown resampling strategies it is necessary to perform a refit
```
Comment (franchuterivera) [inline examples/40_advanced/example_resampling.py:118]: Hi Matthias, what do you mean by unknown resampling strategies?
Do you mean, custom strategies in general, like the ones not listed in the API?

File: examples/40_advanced/example_resampling.py [hunk 2]
```diff
@@ -89,26 +90,39 @@
...
 # It is also possible to use
 # `scikit-learn's splitter classes <https://scikit-learn.org/stable/modules/classes.html#splitter
-# -classes>`_ to further customize the outputs.
+# -classes>`_ to further customize the outputs. In case one needs to have 100% control over the
+# splitting, it is possible to use
+# `scikit-learn's PredefinedSplit <https://scikit-learn.org/stable/modules/generated/
+# sklearn.model_selection.PredefinedSplit.html>`_.
 
+############################################################################
+# Exemplary use of a predefined split. We split the training data by the first
```
Comment (franchuterivera) [inline examples/40_advanced/example_resampling.py:99]: Do you mean: An example of a predefined split?

File: examples/40_advanced/example_resampling.py (hunk 1 above)
Comment (mfeurer) [reply examples/40_advanced/example_resampling.py]: Yes, updated the example.

File: examples/40_advanced/example_resampling.py (hunk 2 above)
Comment (mfeurer) [reply examples/40_advanced/example_resampling.py]: Yes, updated the example.

**Comment by franchuterivera** [comment]
Failures seem to be random, and examples are properly generated. Merging.
